File: 3rd Sem/NM/python/taylorseries.py
def f(x, y): 
    return y - x ** 2 + 1  # The first derivative function y' = f(x, y)

def f_prime(x, y): 
    return 2 * x - 2 * y  # The second derivative function y'' (Assumed form)

def taylor_series_method(x0, y0, h, n):
    x = x0
    y = y0
    print(f"x = {x}, y = {y}")
    
    for i in range(1, n + 1):
        y_prime = f(x, y)  # First derivative (y')
        y_double_prime = f_prime(x, y)  # Second derivative (y'')
        
        # Taylor series method update
        y += h * y_prime + (h ** 2 / 2) * y_double_prime
        x += h
        
        print(f"x = {x}, y = {y}")

# Get initial values, step size, and number of steps from the user
x0, y0 = map(float, input("Enter initial values x0 and y0: ").split())
h = float(input("Enter step size h: "))
n = int(input("Enter number of steps: "))

# Call the Taylor Series method
taylor_series_method(x0, y0, h, n)

File: 3rd Sem/NM/C-program/taylorseries.c
#include <stdio.h>

double f(double x, double y) {
    return y - x * x + 1;  // Function f(x, y) = y - x^2 + 1
}

void taylor_series_method(double x0, double y0, double h, int n) {
    double x = x0;
    double y = y0;
    printf("x = %.6f, y = %.6f\n", x, y);

    for (int i = 1; i <= n; i++) {
        double y_prime = f(x, y);
        double y_double_prime = f(x + h, y + h * y_prime);
        y += h * y_prime + (h * h / 2) * y_double_prime;
        x += h;
        printf("x = %.6f, y = %.6f\n", x, y);
    }
}

int main() {
    double x0, y0, h;
    int n;

    printf("Enter initial values x0 and y0: ");
    scanf("%lf %lf", &x0, &y0);

    printf("Enter step size h: ");
    scanf("%lf", &h);

    printf("Enter number of steps: ");
    scanf("%d", &n);

    taylor_series_method(x0, y0, h, n);

    return 0;
}
